### PR TITLE
Removed logic to get release details from git. Made the script to wor…

### PR DESCRIPTION
**Pull request by bilaharith**
Removed the logic to get release details from git. Made the script to work on 2.5.2 release

### CODE REVIEW

File: patch-cluster-node.sh [hunk 1]
```diff
@@ -41,35 +48,54 @@ do
...
             echo "  -u HDFS_USER    Specifies the user name with superuser privileges on HDFS. Applicable only if the -a switch is specified."
-            echo "  -t TARGET_VERSION "
-            echo "                  Specifies a Release name in the associated Github repo. This release will contain the .jar file to patch."
             echo "  -p DIRECTORY_PREFIX "
             echo "                  Specifies a prefix that is specific to the Hadoop distro & version to search for files to patch."
             echo "  -P HDFS_DIRECTORY_PREFIX "
             echo "                  Specifies a prefix that is specific to the Hadoop distro & version to search on HDFS for files to patch."
             echo "  -R              Rollback installation. Restores previously backed up versions of hadoop-azure jar file. Rollback for HDFS "
-            echo "                  should follow same model as deployment."
+            echo "                  should follow same model as deployment. Specify the backup version for the rollback. Ex: Specify 2020-06-07-10-10-10 "
```
Comment (jamesbak) [inline patch-cluster-node.sh:79]: Need to include the argument value for -R arg

File: patch-cluster-node.sh [hunk 2]
```diff
@@ -41,35 +48,54 @@ do
...
     esac
 done
 
-which jq > /dev/null
-if [ $? -ne 0 ]; then
 
-    echo "This script requires jq to run. Please install using preferred package manager"
-    exit 1
-fi
+[[ "${DIR_PREFIX}" != */ ]] && DIR_PREFIX="${DIR_PREFIX}/"
```
Comment (jamesbak) [inline patch-cluster-node.sh:87]: Not sure this is a change we want. Relying on user to correctly specify the path prefixes degrades the usability of the script and only opens an opportunity for error. 
Consider either hard-coding default values or retaining the properties file mechanism.

File: patch-cluster-node.sh [hunk 3]
```diff
@@ -147,28 +122,6 @@ PATCHED_JAR_FILE_NAME=$(basename $(echo $JAR_ASSET | jq -r '.name') .jar)
 REMOTE_PATCH_PATH=$(echo $JAR_ASSET | jq -r '.browser_download_url')
 LOCAL_PATCH_PATH="/tmp/$PATCHED_JAR_FILE_NAME.new"
 
-# Check for a default properties file for this release. This provides sensible defaults, but can be overridden by commandline args
```
Comment (jamesbak) [inline patch-cluster-node.sh:150]: See earlier comment about retrieving directory prefix values.

File: patch-cluster-node.sh [hunk 4]
```diff
@@ -219,32 +173,30 @@ do
...
 
-        # Backup original file (jar or symlink) if not already backed up
-        if [[ ! -e "${DST}${BACKUP_SUFFIX}" ]]; then
-
-            cp "$DST" "${DST}${BACKUP_SUFFIX}"
-        fi
         # Different handling for symlink or real file
         if [[ ! -h "$DST" ]]; then
 
+            # Backup original file (jar or symlink) if not already backed up
```
Comment (jamesbak) [inline patch-cluster-node.sh]: Consider what happens about making a backup for symlink files.

**Review by jamesbak — CHANGES_REQUESTED** [review]

File: patch-cluster-node.sh (hunk 1 above)
Comment (bilaharith) [reply patch-cluster-node.sh:79]: Done

File: patch-cluster-node.sh (hunk 2 above)
Comment (bilaharith) [reply patch-cluster-node.sh:87]: Currently planning to go ahead with hard coding.

**Review by jamesbak — APPROVED** [review]

**Pull request merged**